### ISSUE TITLE
[4.x] Missing formatting options in rich text editor

### ISSUE THREAD

**Issue by anita-chouhan** [verbb/vizy#307]
### Describe the bug

**Setup:**
- I'm using Vizy fields within matrix blocks.
- Vizy field editor modes are set to _Blocks & Rich Text_.
- Custom config files are being used for editors.
- One custom config offers fewer formatting options than the other.
- The matrix field contains several blocks with Vizy fields.

**Issue:**

If I first add the block with fewer formatting options to the matrix field and then add another block with more options, the second block will only show the common options and not all the available options.

### Steps to reproduce

Here are the instructions:

1. Create a matrix field and add two blocks to it.
2. Name the blocks "Rich Text" and "List" respectively.
3. Add a Vizy-type field to each block.
4. Set the editor mode to "Blocks and Rich Text".
5. In the editor config option, select the "RichText" and "List" config files for the fields.
6. Enable the empty paragraph removal option for both blocks.
7. Add the matrix field to an entry.
8. Go to the entry and select the "List" block first, then choose the "Rich Text" block.
9. Check the available formatting options for both blocks.

Config files to test:
**RichText.json**
`{
  "buttons": [
    "formatting",
    "bold",
    "italic",
    "link",
    "unordered-list",
    "ordered-list"
  ],
  "toolbarFixed": true,
  "formatting": [
    "h3",
    {
      "title": "Heading 3 Caption",
      "type": "heading",
      "editorStyle": ".vizy-h3-caption { font-size: 1.3em; }",
      "attrs": { "class": "vizy-h3-caption", "level": "6" }
    },
    "h4",
    "h5",
    "blockquote",
    "paragraph"
  ]
}
`

**List.json** 
`{
  "buttons": ["formatting", "bold", "italic", "link", "unordered-list"],
  "formatting": ["h3", "paragraph"]
}`
 

### Craft CMS version

4.10.6

### Plugin version

2.1.20

### Multi-site?

No

**Comment by engram-design** [comment]
I've just replicated your field setup and that seems to be working.

![CleanShot 2024-08-09 at 12 57 48@2x](https://github.com/user-attachments/assets/99e28f86-0a21-4e22-b3c7-dab924fa3ca4)

Anything else worth noting about your reproduction?

**Comment by anita-chouhan** [comment]
> I've just replicated your field setup and that seems to be working.
> 
> ![CleanShot 2024-08-09 at 12 57 48@2x](https://private-user-images.githubusercontent.com/1221575/356445339-99e28f86-0a21-4e22-b3c7-dab924fa3ca4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0NDU2NzIsIm5iZiI6MTcyMzQ0NTM3MiwicGF0aCI6Ii8xMjIxNTc1LzM1NjQ0NTMzOS05OWUyOGY4Ni0wYTIxLTRlMjItYjNjNy1kYWI5MjRmYTNjYTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTJUMDY0OTMyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjkwODE2NTM4N2VmNjFmZGVlMGE5Yzc4NjAwMzYxNzRlYjNjNjZkOGM5NGIwZmQ4ODVkNzEyZjU2N2FjMjEwMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.W9qxXrIN4MBt6xmHL5G5V1jag0luYFHZ9yZsoBU0MIo)
> 
> Anything else worth noting about your reproduction?

@engram-design I expect the _Rich Text_ editor block to show all the formatting options available in the **RichText.json** config. Currently, the _H4, H5_ and _Blockquote_ are missing.

**Comment by engram-design** [comment]
Oh, my mistake! I missed those due to the JSON formatting.

```json
{
    "buttons":[
        "formatting",
        "bold",
        "italic",
        "link",
        "unordered-list",
        "ordered-list"
    ],
    "toolbarFixed":true,
    "formatting":[
        "h3",
        {
            "title":"Heading 3 Caption",
            "type":"heading",
            "editorStyle":".vizy-h3-caption { font-size: 1.3em; }",
            "attrs":{
                "class":"vizy-h3-caption",
                "level":"6"
            }
        },
        "h4",
        "h5",
        "blockquote",
        "paragraph"
    ]
}
```

Indeed, `h4`, `h5` and `blockquote` are missing. Fixed for the next release. To get this early, run `composer require verbb/vizy:"dev-craft-4 as 2.1.20"`.

**Comment by anita-chouhan** [comment]
Hello @engram-design, thank you for looking into it. I tried the bug-fix release but found that formatting options are not working. The only two options that are working are _Heading 3 Caption_ and _Link_ 

https://github.com/user-attachments/assets/0be0fe14-ff61-4314-a8ee-50eff8c23c51

.

**Comment by engram-design** [comment]
Should be fixed in [2.1.22](https://github.com/verbb/vizy/releases/tag/2.1.22)
